samples/99_zil_interpreter/tests/builtins_tests.rb

```diff
@@ -28,8 +28,141 @@ def test_builtin_lval_raises_error_when_not_existing(args, assert)
 end
+
+def test_builtin_minus(args, assert)
+  zil_context = build_zil_context(args)
+
+  # <- 10 1 <- 5 1>>
+  result = zil_context.globals[:-].call [10, 1, Syntax::Form.new(:-, 5, 1)], zil_context
+  assert.equal! result, 5
+
+  # <- 5>
+  result = zil_context.globals[:-].call [5], zil_context
+  assert.equal! result, -5
+
+  # <- >
+  result = zil_context.globals[:-].call [], zil_context
+  assert.equal! result, 0
+end
+
+def test_builtin_multiply(args, assert)
+  zil_context = build_zil_context(args)
+
+  # <* 1 5>
+  result = zil_context.globals[:*].call [1, 5], zil_context
+  assert.equal! result, 5
+
+  # <* 1 5 <* 5 2>>
+  result = zil_context.globals[:*].call [1, 5, Syntax::Form.new(:*, 5, 2)], zil_context
+  assert.equal! result, 50
+
+  # <* 1>
+  result = zil_context.globals[:*].call [], zil_context
+  assert.equal! result, 1
+end
+
+def test_builtin_divide(args, assert)
+  zil_context = build_zil_context(args)
+
+  # </ 10 </ 50 10>>
+  result = zil_context.globals[:/].call [10, Syntax::Form.new(:/, 50, 10)], zil_context
+  assert.equal! result, 2
+
+  # </ 10 2>
+  result = zil_context.globals[:/].call [10, 2], zil_context
+  assert.equal! result, 5
+
+  # </ 5>
+  result = zil_context.globals[:/].call [5], zil_context
+  assert.equal! result, 0
+
+  # </>
+  result = zil_context.globals[:/].call [], zil_context
+  assert.equal! result, 1
+
+  # </ 5.0>
+  result = zil_context.globals[:/].call [5.0], zil_context
+  assert.equal! result, 0.2
+
+  # </ 1.5 0.5>
+  result = zil_context.globals[:/].call [1.5, 0.5], zil_context
+  assert.equal! result, 3.0
+
+  # </ 11 7 2.0>
+  result = zil_context.globals[:/].call [11, 7, 2.0], zil_context
+  assert.equal! result, 0.5
+end
+
+def test_builtin_min(args, assert)
+  zil_context = build_zil_context(args)
+
+  # <MIN 1>
+  result = zil_context.globals[:MIN].call [1], zil_context
+  assert.equal! result, 1
+
+  # <MIN 1.0>
+  result = zil_context.globals[:MIN].call [1.0], zil_context
+  assert.equal! result, 1.0
+
+  # <MIN 2 3>
+  result = zil_context.globals[:MIN].call [2, 3], zil_context
+  assert.equal! result, 2
+
+  # <MIN 2.0 3>
+  result = zil_context.globals[:MIN].call [2.0, 3], zil_context
+  assert.equal! result, 2.0
+
+  # <MIN 3 4 <MIN 5 6.0>>
+  result = zil_context.globals[:MIN].call [3, 4.0, Syntax::Form.new(:MIN, 5, 6.0)], zil_context
+  assert.equal! result, 3
+
+  # <MIN>
+  zil_context.globals[:MIN].call [], zil_context
+  raise 'No exception occurred when invoking MIN with no arguments!'
+rescue FunctionError
+  assert.ok!
+end
+
+def test_builtin_random(args, assert)
+  zil_context = build_zil_context(args)
+
+  # <RANDOM 1>
+  result = zil_context.globals[:RANDOM].call [1], zil_context
+  assert.equal! result, 1
+
+  # <RANDOM 2>
+  result = zil_context.globals[:RANDOM].call [2], zil_context
+  assert.true! result >= 1 && result <= 2
+
+  # <RANDOM 3>
+  result = zil_context.globals[:RANDOM].call [3], zil_context
+  assert.true! result >= 1 && result <= 3
```

Comment:
some rather small point - but technically it would be better to do something like
```rb
results = 10.times.map { zil_context.globals[:RANDOM].call([1], zil_context) }
assert.true! results.all? { |result| result == 1 }, '<RANDOM 1> returned a number other than 1'
```

Reply:
Sounds good.  Fixed.